modules/ensemble/lib/widget/collapsible.dart

```diff
@@ -0,0 +1,345 @@
+import 'package:ensemble/framework/error_handling.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/view/data_scope_widget.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:flutter/material.dart';
+import 'package:accordion/accordion.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:yaml/yaml.dart';
+import 'package:ensemble/framework/widget/icon.dart' as ensembleIcon;
+
+class HeaderStyle {
+  Color? backgroundColor;
+  Color? backgroundColorOpened;
+  Color? borderColor;
+  Color? borderColorOpened;
+  double? borderWidth;
+  double? borderRadius;
+  EdgeInsets? padding;
+
+  HeaderStyle({
+    this.backgroundColor,
+    this.backgroundColorOpened,
+    this.borderColor,
+    this.borderColorOpened,
+    this.borderWidth,
+    this.borderRadius,
+    this.padding,
+  });
+}
+
+class ContentStyle {
+  Color? backgroundColor;
+  Color? borderColor;
+  double? borderWidth;
+  double? borderRadius;
+  double? horizontalPadding;
+  double? verticalPadding;
+
+  ContentStyle({
+    this.backgroundColor,
+    this.borderColor,
+    this.borderWidth,
+    this.borderRadius,
+    this.horizontalPadding,
+    this.verticalPadding,
+  });
+}
+
+class Collapsible extends StatefulWidget
+    with
+        Invokable,
+        HasController<CollapsibleController, CollapsibleState> {
+  static const type = 'Collapsible';
+
+  Collapsible({Key? key}) : super(key: key);
+
+  final CollapsibleController _controller = CollapsibleController();
+
+  @override
+  CollapsibleController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => CollapsibleState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'openSections': () => _controller.openSections,
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'items': (value) => _controller.items = List<YamlMap>.from(value),
```

Comment:
Why don't you just use "children" property? See the other containers. This will ensure the children won't get evaluated until it is ready to be used. This may not work properly.

Reply:
For now, i have tested its working fine but I have seen dropdown example, how it is using children and itemtemplate, I will create one more ticket and PR against it, enhancing it to support itemtemplate as well.

Reply:
no you want to address this now. Once the schema is out harder to adjust. Also the real reason is children and itemTemplate are special such taht their content won't be evaluated immeidately. You want to only evaluate the content when you are ready to render it